### DISCUSSION TITLE
Pulling in Data from RSS Feeds

### DISCUSSION THREAD

**Discussion by ghost** [strapi/strapi#7772] [Legacy Topics]
I would like some advice on how to approach the following using strapi.

I want to create a form or field in the backend where I can enter in an RSS feed and have certain elements of that feed saved to the database as an entry for a News Article content type.  I then want to be able to query that feed every X number of minutes for new articles and have those also saved to the database.

I am wondering -- what is the best way to approach this with Strapi?  To create a custom plugin?  Something else?  

Thanks.

**Comment by derrickmehaffy** [comment]
I'm not sure if it would be the best fit to directly put this into Strapi, especially if you plan to scale your application (horizontally or vertically).

The main problem would be if you are running this on a loop or a cron, and during scale you would have multiple instances trying to fetch, parse, and update the database with the new data.

My suggestion in this case is you would want to build a custom `node-cron` script that can fetch the RSS feed, parse it, then fire the data off to Strapi to store. This would give you a single "source of truth". You could for example have a model where you place the links to the RSS feeds, in which the node-cron can fetch, parse, loop through, and send the data back to Strapi (over REST or GraphQL mutation). But directly building in the cron isn't something I would recommend.

If you don't plan to scale you can run the cron with the built in cron option documented here: https://strapi.io/documentation/v3.x/concepts/configurations.html#cron-tasks
